### ISSUE TITLE
quirk: ACL doesn't create new file after existing log file is deleted

### ISSUE THREAD

**Issue by i3roly** [Talryn/AutoCombatLogger#6]
maybe this is an interface issue.

i have had this 'problem' for a while. 

if i do a bunch of runs and then use warcraftlogs to upload and then delete the logfile, ACL will not create a new log file for the runs that follow.

so pretty much every run after the file is deleted (while wow is running) is 'lost'. i *think* if i remember to type /combatlogging (or whatever) after deleting the file, it will show up, but i was hoping maybe there's a way for your addon to check if the file exists, and if not to create it.

thanks for your good work.

**Comment by Talryn** [comment]
Unfortunately, I may not be able to fix your issue in the addon.  The WoW API only allows starting or stopping of combat logging.  The addon doesn't know anything about files.  It is isolated from the rest of the computer.  When ACL toggles combat logging it is the same as when you type `/combatlog`.

Normally if you leave and/or enter a raid or zone, it would re-check if it should enable or disable combat logging.  Are you remaining inside the zone?  I'm wondering what is happening and what the addon is seeing.

If you are in the zone and doing a reset, I could check if that can be detected and then re-check combat logging.  I'm also curious if when you delete the log file, does it show combat logging as disabled?  The minimap icon or the LDB icon display what the addon believes is the current state of combat logging.

After deleting the log, perhaps try a `/reload` and see if it correctly handles it.  That will reload the client and addons.  ACL will re-check combat logging at that point.  If a reload doesn't solve your issue, then I don't think any chance in the addon will fix it.

It is also possible that deleting the file while the game is running is always going to cause issues.

**Comment by i3roly** [comment]
Hi,

Good questions.

1. No I'm not remaining inside the zone.

2. When I delete the log file, nothing unusual is reported. This is why I filed the issue. CombatLogger's prompts are exactly what they should be..

 I think if I toggle /combatlog manually, a new log file appears, but I have to double check. I never see that 'yellow text' when ACL starts and stops logging, only when I start and stop it manually.

 I have not tried reloading but this isn't a bad idea either‎.

You'd think the game checks if the file exists before trying to log data there. I'll check on whether doing it manually changes behaviour.

bless,
g
From: Talryn
Sent: Tuesday, 28 February 2023 11:13 AM
To: Talryn/AutoCombatLogger
Reply To: Talryn/AutoCombatLogger
Cc: gagan sidhu; Author
Subject: Re: [Talryn/AutoCombatLogger] quirk: ACL doesn't create new file after existing log file is deleted (Issue #6)


Unfortunately, I may not be able to fix your issue in the addon. The WoW API only allows starting or stopping of combat logging. The addon doesn't know anything about files. It is isolated from the rest of the computer. When ACL toggles combat logging it is the same as when you type /combatlog.

Normally if you leave and/or enter a raid or zone, it would re-check if it should enable or disable combat logging. Are you remaining inside the zone? I'm wondering what is happening and what the addon is seeing.

If you are in the zone and doing a reset, I could check if that can be detected and then re-check combat logging. I'm also curious if when you delete the log file, does it show combat logging as disabled? The minimap icon or the LDB icon display what the addon believes is the current state of combat logging.

After deleting the log, perhaps try a /reload and see if it correctly handles it. That will reload the client and addons. ACL will re-check combat logging at that point. If a reload doesn't solve your issue, then I don't think any chance in the addon will fix it.

It is also possible that deleting the file while the game is running is always going to cause issues.

**Comment by i3roly** [comment]
yeah i thought toggling /combatlog would bring the log back, but i guess not.

so it may be what you're saying: deleting the log while hte game is running is a bad idea.

i'm surprised the WoW UI doesn't check to see if the file still exists before writing to it.

the riddle becomes: where the fuck is it writing all of this data to? it has to be somewhere.

**Comment by Talryn** [comment]
My setup is different but I was able to confirm that deleting a log file does result in the same issue you are seeing.  I didn't get a chance to test all the scenarios though.

Have you tried WarcraftLogs archive feature?  I wonder if that helps versus deleting it.  I haven't been posting logs on WCL lately but when I did, I did keep the files for a while before deleting them so I never tried to do what you are doing.  I also only dealt with logs after the raid was over so I never ran into the issue.

I believe the WoW client is just silently failing after you delete the log file.  It would be nice if it turned logging off but apparently it does not.  There are two possibilities though.  If you delete it but it goes to the trash, perhaps it is still writing there, otherwise it is just failing and not reporting it.

Have you checked what happens if you disable combat logging (and verify it is off), delete the log, then continue playing?  I'm not sure if that would resolve the issue.  I assume it may be happening already based on what you said but worth checking.

Unfortunately, I think only Blizzard can fix this issue.  You may need to wait on deleting logs unless there is a workaround we haven't found yet.
